lib/rules/constructor-super.js

```diff
@@ -0,0 +1,472 @@
+            for (var i = 0; i < keys.length; ++i) {
+                var child = node[keys[i]];
+                if (Array.isArray(child)) {
+                    child.forEach(this.visit, this);
+                } else {
+                    this.visit(child);
+                }
+            }
+        }
+    }
+};
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+
+    /**
+     * Searches a class node where the current context is.
+     *
+     * If `needsCheckInConstructor` is true, if the current context is not in constructor, this returns `null` always.
+     *
+     * @param {boolean} needsCheckInConstructor - Whether or not the current context is in constructor.
+     * @returns {ClassDeclaration|ClassExpression|null} the found class node, or `null`.
+     */
+    function getClassInAncestor(needsCheckInConstructor) {
+        var ancestors = context.getAncestors();
+        var inConstructor = !needsCheckInConstructor;
+        for (var i = ancestors.length - 1; i >= 0; --i) {
+            var ancestor = ancestors[i];
+            switch (ancestor.type) {
+                case "ClassDeclaration":
+                case "ClassExpression":
+                    return inConstructor ? ancestor : null;
+
+                case "MethodDefinition":
+                    if (ancestor.kind === "constructor") {
+                        inConstructor = true;
+                    }
+                    break;
+
+                default:
+                    break;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Analyzes execution paths of a node and counts `super()` callings.
+     *
+     * @param {MethodDefinition} node - A node to analyze.
+     * @returns {{min: number, max: number}} A number of `super()` calling. `min` is the minimum number in all paths. `max` is the maximum number in all paths.
+     */
+    function countSuperCalling(node) {
+        return Object.create(analyzerProto).analyze(node);
+    }
+
+    return {
+        // if removes `:exit`, the `parent` property of descendent nodes becomes `undefined`.
+        "MethodDefinition:exit": function(node) {
+            if (node.kind !== "constructor") {
+                return;
+            }
+            var classNode = getClassInAncestor(false);
+            if (classNode == null || classNode.superClass === null || isNullLiteral(classNode.superClass)) {
+                return;
+            }
+
+            var count = countSuperCalling(node);
+            if (count && count.max >= 2) {
+                context.report(node.key, "this constructor has redundant `super()` callings.");
+            } else if (count && count.min === 0) {
+                context.report(node.key, "this constructor should call `super()`.");
+            }
+        },
+
+        "Super:exit": function(node) {
```

Comment:
I upgraded estraverse. 
The diff is here: https://github.com/mysticatea/eslint/commit/487b36fcd2e0b287d70908d6fdce1e8f68933252